modules/transactions/src/main/java/org/apache/ignite/internal/tx/TxManager.java

```diff
@@ -146,6 +147,22 @@ CompletableFuture<Void> cleanup(
             @Nullable HybridTimestamp commitTimestamp
     );
 
+    /**
+     * Sends unlock request to the primary nodes of each one of {@code partitions}.
+     *
+     * @param partitions Enlisted partition groups.
+     * @param commit {@code true} if a commit requested.
+     * @param commitTimestamp Commit timestamp ({@code null} if it's an abort).
+     * @param txId Transaction id.
+     * @return Completable future of Void.
+     */
+    CompletableFuture<Void> unlock(
+            Collection<TablePartitionId> partitions,
```

Comment:
The partitions collection may be nullable (in case of recovery), it should be marked here